tools/spanner/purge_ttl/src/main.rs

```diff
@@ -0,0 +1,147 @@
+use std::env;
+use std::error::Error;
+use std::net::{IpAddr, UdpSocket};
+use std::sync::Arc;
+use std::time::Instant;
+
+use log::{info, trace, warn};
+use cadence::{
+    BufferedUdpMetricSink, Metric, QueuingMetricSink, StatsdClient, Timed, DEFAULT_PORT,
+};
+use googleapis_raw::spanner::v1::{
+    spanner::{BeginTransactionRequest, CreateSessionRequest, ExecuteSqlRequest, Session},
+    spanner_grpc::SpannerClient,
+    transaction::{TransactionOptions, TransactionOptions_PartitionedDml, TransactionSelector},
+};
+use grpcio::{CallOption, ChannelBuilder, ChannelCredentials, EnvBuilder, MetadataBuilder};
+
+pub struct MetricTimer {
+    pub client: StatsdClient,
+    pub label: String,
+    pub start: Instant,
+}
+
+impl Drop for MetricTimer {
+    fn drop(&mut self) {
+        let lapse = (Instant::now() - self.start).as_millis() as u64;
+        match self.client.time(&self.label, lapse) {
+            Err(e) => {
+                warn!("⚠️ Metric {} error: {:?}", self.label, e);
+            }
+            Ok(v) => {
+                info!("⌚ {:?}", v.as_metric_str());
+            }
+        }
+    }
+}
+
+pub fn start_timer(client: &StatsdClient, label: &str) -> MetricTimer {
+    trace!("⌚ Starting timer... {:?}", label);
+    MetricTimer {
+        start: Instant::now(),
+        label: label.to_owned(),
+        client: client.clone(),
+    }
+}
+
+pub fn statsd_from_env() -> Result<StatsdClient, Box<dyn Error>> {
+    let statsd_host = env::var("STATSD_HOST")
+        .unwrap_or("localhost".to_string())
```

Comment:
IpAddr won't accept a hostname

```suggestion
        .unwrap_or("127.0.0.1"to_string())
```